### ISSUE TITLE
End-of-cooldown and global cooldown overlapping behaviour

### ISSUE THREAD

**Issue by void-ow** [WeakAuras/WeakAuras2#5097]
Hello, sorry, I'm going to be a horrible person that says "I liked the way things worked before"
TLDR: I'm referring to this commit - https://github.com/InfusOnWoW/WeakAuras2/commit/ee20b5d9fd41baf17aadc95e5454dd292a00cf8c

**Is your feature request related to a problem? Please describe.**
WA spell cooldown tracking until 5.12.1 used to consider spells being "on gcd cooldown" if a spell would come off cooldown during the gcd window. This resulted in a functionality that I personally liked:
1. my cooldown tracker aura was set to de-saturate a spell on cooldown
2. if a spell would come off cooldown during the period of a gcd it would light up prematurely - starting from the beginning of a gcd

If, for example, I'm casting Smite and my Mind Blast would come off cooldown at any point of time during the cast, Mind Blast would light up the moment I start casting Smite, giving me the time to plan my rotation ahead of the time.

In the latest version of WA though, my cooldown trackers light up at the exact moment the spell is off cooldown, which is technically more correct but can sometimes lead to frustration (at least for me, because I'm used to the previous behavior)

**Describe the solution you'd like**
I'd like to propose some sort of additional opt-in only configuration parameter for cooldown tracker weakauras (for example "predictive cooldown") that would retain the previous, before-5.12.1 functionality

**Describe alternatives you've considered**
The way I've been handling this existential crisis myself for the past year is by manually reverting the aforementioned commit in my WA distribution.

**Additional context**
I totally understand if such a request is out of scope for you guys, and I'm fine with manually modifying my own version of WA forever if that's the case, however annoying it may be.
I'm making this ticket in hopes I'm not alone in preferring the old behavior and that my suggestion can be a legitimate feature.

Thank you.

**Comment by InfusOnWoW** [comment]
Duplicate of https://github.com/WeakAuras/WeakAuras2/issues/5666
